src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -104,17 +107,66 @@ private void UpdateNextRequestUri(ResponseHeaders headers)
             switch (_headerSource)
             {
                 case HeaderSource.OperationLocation when headers.TryGetValue("Operation-Location", out string? operationLocation):
-                    _nextRequestUri = operationLocation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(operationLocation, _startRequestUri);
                     return;
                 case HeaderSource.AzureAsyncOperation when headers.TryGetValue("Azure-AsyncOperation", out string? azureAsyncOperation):
-                    _nextRequestUri = azureAsyncOperation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(azureAsyncOperation, _startRequestUri);
                     return;
                 case HeaderSource.Location when hasLocation:
-                    _nextRequestUri = location!;
+                    _nextRequestUri = AppendOrReplaceApiVersion(location!, _startRequestUri);
                     return;
             }
         }
 
+        internal static string AppendOrReplaceApiVersion(string uri, Uri startRequestUri)
+        {
+            if (TryGetApiString(startRequestUri, out ReadOnlySpan<char> apiVersion))
```

Comment:
Since this is only ever called with _startRequestUri can we do this as a lazy loaded value so that we only get the apiversion once?

Reply:
The AppendOrReplaceApiVersion method is first called before the creation of a NextLinkOperationImplementation instance. I added _apiVersion property to store the value in constructor from the result of the first call and also refactored AppendOrReplaceApiVersion to take in apiVersion directly. Therefore, TryGetApiString is only called once now.